content/blog/2022/100-percent-test-coverage.md

```diff
@@ -0,0 +1,157 @@
+can also enable you to write more detailed error messages when you make a
+mistake, acting as an error library.
+
+In some cases, tests can prove that big refactors actually work; it all depends
+on how they are structured. Very mock-heavy, small unit tests tend to require
+rewriting if the implementation changes. On the other hand a good integration
+test shouldn't need changing unless the requirements have changed.
+
+Finally, having robust test coverage can help you understand features of a
+codebase or library before writing a line of code. Tests can be a great guide to
+the ways certain features work under the hood.
+
+There are also cons to 100% test coverage:
+
+One thing that 100% test coverage can lead to is an unwillingness to make big
+refactors that you know will impact your test suite. It also might impact how
+you write your code in the first place -- for example, trying to write code that
+will be less arduous to test.
+
+With unit tests, mocking leads to fragile tests that may not accurately
+represent your codebase. I personally hate the process of mocking a complicated
+piece of code, and it is often difficult to find documentation and examples for
+your exact esoteric piece of code. You’ll often need to mock to cover all
+aspects of the codebase, so it’s a big problem for higher test coverage.
+
+Relatedly, having 100% test coverage can lead to misplaced confidence in your
+test suite. If you are relying on unit tests and implementation tests solely to
+check your codebase, things can easily be missed. Changes to the visual
+presentation that could be caught by regression tests are often missed in big
+refactors when relying solely on unit tests, for example. Implementation tests
+can give false confidence about things that they cover; end to end tests may
+pick up broken UI or edge cases that wouldn’t come up in a partial test of the
+system.
+
+Getting to 100% test coverage adds extra time to development processes, as it
+takes time to implement and change tests. This slows down creation of features
+when chasing the last percentage points of coverage. They can also be an
+impediment to big refactors when APIs and tests are too tightly coupled. More
+tests also make for a slower test suite, which can slow the development process.
+
+Another issue is overcoming inertia to start writing them, especially when the
+tests are left to the end. It’s often easier to use a test driven approach
+without a code coverage target; writing tests that are embedded in the code
+writing process means they cannot be skipped, and not having a target means that
+there is less infilling to do at the end of writing code.
+
+## Conclusion
+
+Test coverage is one tool in our toolbox, and having a metric to measure it
+gives some reassurance that we have looked at the code from many angles. The
+other testing dimensions are covered in our testing processes -- we often test
```

Comment:
The conclusion looks great! 🎉  One suggestion here is to break up this sentence that starts with `The other testing dimensions...` - it was a bit hard to follow since it was quite long.